build_runner/test/generate/build_test.dart

```diff
@@ -123,7 +123,7 @@ void main() {
     var cachedGraph = new AssetGraph.deserialize(
         JSON.decode(writer.assets[graphId].stringValue));
 
-    var expectedGraph = new AssetGraph();
+    var expectedGraph = new AssetGraph.build([], new Set());
```

Comment:
Perhaps... but hopefully not. I think only tests would use it so I want to take a shot at finding a new pattern for tests that doesn't require manual graph manipulation like this - or maybe subclassing the graph just for tests.